.github/workflows/test.yml

```diff
@@ -6,8 +6,13 @@ jobs:
   test:
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        tag: [latest, dev]
```

Comment:
Current stable of `2.10` is probably the last version without NNDB. I think we should have it pinned as well as `latest` &  `dev` tags. What do you think?

Ping @exercism/dart.

Reply:
Not sure why there is a need to pin `2.10` even if it is the last version without NNDB
Pinning versions in CI will lead to inability to use new features in code in the future, there will be a need to pin next version and so on

Reply:
@devkabiir I'm fine either way you want to do it. As long as we would know our code works pre-NNDB, then I'm fine not pinning it.

But I also see the benefit of ensuring we are backwards compatible with pre-NNDB code (for the exercise code).

I believe NNDB is slated for version 2.12. So if we pin 2.10, then it will definitely have to be updated to 2.11 once 2.11 comes out.

Or we could wait until 2.12 is released and then go back and pin 2.11.

Any way you want to do it is good to me.

Reply:
I can somewhat confidently say that the current code is compatible with `2.10`.
I don't know if `2.11` will be a real release (probably with breaking changes) or skipped over.
I know for certain `2.12` will cause issue in current code, dependencies, exercises, and solutions(example and submitted ones).

Refactoring will certainly be required for `2.12`. By convention, you don't refactor without enough tests to back you up. 
A pinned `2.10` will serve as a basic test case as well as a working state which also allows reverting to this commit via git as quickly as possible should the need arise.

We can also choose to include `// @dart=2.9` at the beginning of dart files to _opt-out_ of NNBD.  Although that's just delaying the inevitable refactoring.

NNBD will cause a lot of errors in non-migrated code for exercises, solutions, and this repo & it's not released yet. NNBD is also marked as a breaking change.  We aren't even running `dart analyze`  and don't have any preference over `analysis_options.yaml` for submitted solutions so we may have more issues in our code than we can anticipate. People solving exercises 6 months from now will most certainly be using post-NNBD dart, which might require us to include a few NNBD-specific lint rules. We also have #171, although I think it would be best to work on that post-NNBD

It's true that pinning versions prevents using new features, but it also prevents bugs to sneak-in as well as PRs that don't make changes to dart code from suddenly being broken (because the `latest` tag now points to `2.12` instead of `2.10`).

So my ideal scenario would be
 - Pin pre-NNBD stable release (currently `2.10`)
 - Pin post-NNBD stable release (not available yet)
 - Pin `latest` (once the latest tab points to `2.12`)
 - Pin `dev` (doesn't matter if this fails in CI)

I honestly think this is better suited as a separate discussion. For the purposes of this PR, I'd prefer `2.10` be pinned.
If it's possible to pin to something like `^2.10.0` then that has precedence. Since it's less likely to be containing a breaking change.